lucene/facet/src/test/org/apache/lucene/facet/taxonomy/TestTaxonomyFacetCounts.java

```diff
@@ -711,6 +723,11 @@ public void testRandom() throws Exception {
         }
       }
 
+      // Test facet labels for each matching test doc
+      List<List<FacetLabel>> actualLabels = getTaxonomyFacetLabels(tr, config, fc);
+      assertEquals(expectedLabels.size(), actualLabels.size());
```

Comment:
Hmm I think `expectedLabels` filters out empty `List<FacetLabel>` but `actualLabels` does not, so this might false trip?

Reply:
Nice catch, thanks.  I fixed `actualLabels` generation in `FacetTestCase.getAllTaxonomyFacetLabels()` method to filter out empty `List<facetLabels>`.

Reply:
Hmm I don't like filtering out "no facet fields" case from `FacetTestCase.getAllTaxonomyFacetLabels()` -- that means callers cannot safely map hits to the `List<FacetLabel>`?

Could we instead keep the empty lists in both expected and actual?

Reply:
Done. I made changes to keep empty lists in both `expectedLabels` and `actualLabels` and also updated javadoc for `FacetTestCase.getAllTaxonomyFacetLabels()` to reflect this for `actualLabels`